test/e2e/e2e_suite_test.go

```diff
@@ -24,118 +24,98 @@ import (
 	"fmt"
 	"io/ioutil"
 	"os"
-	"os/exec"
-	"path"
 	"testing"
-	"time"
-
-	"github.com/onsi/ginkgo"
-	"github.com/onsi/gomega"
-	appsv1 "k8s.io/api/apps/v1"
-	apimachinerytypes "k8s.io/apimachinery/pkg/types"
-	"sigs.k8s.io/cluster-api/test/helpers/kind"
-	"sigs.k8s.io/cluster-api/test/helpers/scheme"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/cluster-api/test/framework/generators"
+	"sigs.k8s.io/cluster-api/test/framework/management/kind"
 	"sigs.k8s.io/cluster-api/util"
-	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 func TestE2E(t *testing.T) {
-	gomega.RegisterFailHandler(ginkgo.Fail)
-	ginkgo.RunSpecs(t, "e2e Suite")
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Cluster API E2E Suite")
 }
 
 const (
 	capiNamespace      = "capi-system"
+	capiTestNamespace  = "default"
```

Comment:
Should we use a randomized name here instead of the default namespace? I worry this might make parallelizing or running the tests against an existing environment more difficult.

Reply:
@detiber Were you thinking of a `namespace` per `It` test block or just for this suite of tests?

Reply:
@wfernandes definitely not per `It`, but maybe a bit more granular than per-suite as well.